static/js/components/ClassificationForm.jsx

```diff
@@ -56,197 +59,229 @@ const allowed_classes = (hierarchy) => {
 
 const ClassificationForm = ({ obj_id, taxonomyList }) => {
   const classes = useStyles();
+  const dispatch = useDispatch();
+  const groups = useSelector((state) => state.groups.userAccessible);
+  const groupIDToName = {};
+  groups.forEach((g) => {
+    groupIDToName[g.id] = g.name;
+  });
+  const ITEM_HEIGHT = 48;
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: ITEM_HEIGHT * 4.5,
+        width: 250,
+      },
+    },
+  };
+
   const latestTaxonomyList = taxonomyList.filter((t) => t.isLatest);
 
-  function reducer(state, action) {
-    switch (action.name) {
-      case "taxonomy_index":
-        return {
-          ...state,
-          [action.name]: action.value,
-          allowed_classes: allowed_classes(
-            latestTaxonomyList[action.value].hierarchy
-          ),
-        };
-      default:
-        return {
-          ...state,
-          [action.name]: action.value,
-        };
+  const handleSubmit = async ({ formData }) => {
+    // Get the classification without the context
+    const classification = formData.classification.split(" <> ")[0];
```

Comment:
Why not get the form to send the value you are interested in instead?

Reply:
I was getting validation errors related to duplicate values in the schema enum because there are classifications with duplicate names (but different contexts) so this was my workaround.

Reply:
OK, so this is slightly bizarre: why is freeform text allowed for classification? 

https://skyportal.io/docs/api.html#/paths/~1api~1classification/post

Should we not validate that classifications are from an existing taxonomy?

Reply:
That's more of an issue with the docs I think. There is a check for allowed classifications in the handler code (and the frontend form only allows selection from the allowed classes)

Reply:
https://github.com/skyportal/skyportal/blob/fe49f4b99fadefb7be08769dab7f63a5a8640ba9/skyportal/handlers/api/classification.py#L123-L136